packages/react-server/core/renderMiddleware.js

```diff
@@ -894,8 +904,8 @@ function setupLateArrivals(res) {
 	});
 
 	// TODO: maximum-wait-time-exceeded-so-cancel-pending-requests code
-	var promises = notLoaded.map( result => result.entry.dfd.promise );
-	RLS().lateArrivals = Q.allSettled(promises)
+	var promises = notLoaded.map( result => result.entry.promise );
+	RLS().lateArrivals = Promise.all(promises).catch(() => {})
```

Comment:
Updated to [a correct port](https://github.com/redfin/react-server/pull/62/commits/c3f1756d9199cba2993e49c92766cea182c9bb65) of `Q.allSettled()` using [`promise.reflect`](http://bluebirdjs.com/docs/api/reflect.html).

Not going to change the error handling behavior here.  Trying to stick to a straight port.  Goal is no change in behavior (aside from performance increase :rocket:).